test.sh

```diff
@@ -1,32 +1,62 @@
 #!/bin/sh
 
+RED="\033[0;31m"
+GREEN="\033[0;32m"
+NC="\033[0m"
+
+CHECK="\xE2\x9C\x94"
+X="\xE2\x9D\x8C"
+
+testDay () {
+    day=$1
+    fileNo=$2
+    subcmd=""
+    fileSuffix=""
+    if [ "$fileNo" = 0 ]; then
+        echo "Testing Day $day real"
+    elif [ "$fileNo" = 1 ]; then
+        echo "Testing Day $day test"
+        subcmd="test"
+        fileSuffix="_test"
+    else
+        echo "Testing Day $day test $fileNo"
+        subcmd="test $fileNo"
+        fileSuffix="_test$fileNo"
+    fi
+    # Run the test in subshell with echoing enabled (the "set -x" part)
+    # fixes both "set +x" having to be echoed and "set +x" overriding return code from test
+    # https://stackoverflow.com/a/19392242
+    ( set -x; ./run.sh "$day" $subcmd | diff "$day/expected$fileSuffix" - )
```

Comment:
The output from set -x appears to be flaky on the Ubuntu that GitHub Actions uses.